core/controllers/feedback_test.py

```diff
@@ -554,3 +556,216 @@ def test_feedback_threads_with_suggestions(self):
+
+        self.topic_id = topic_services.get_new_topic_id()
+        self.save_new_topic(
+            self.topic_id, self.owner_id, 'Name', 'Description',
+            [], [], [], [], 1)
+
+    def test_get_feedback_threads_with_new_structure_editors_disabled_raise_404(
+            self):
+        self.login(self.OWNER_EMAIL)
+
+        feedback_services.create_thread(
+            feconf.ENTITY_TYPE_TOPIC, self.topic_id, self.owner_id,
+            'a subject', 'some text', has_suggestion=True)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_EDITORS', False):
+            self.get_json(
+                '%s/%s' % (
+                    feconf.FEEDBACK_THREADLIST_URL_PREFIX_FOR_TOPICS,
+                    self.topic_id), expected_status_int=404)
+        self.logout()
+
+    def test_get_feedback_threads_linked_to_topics(self):
+        self.login(self.OWNER_EMAIL)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_EDITORS', True):
+            response_dict = self.get_json(
+                '%s/%s' % (
+                    feconf.FEEDBACK_THREADLIST_URL_PREFIX_FOR_TOPICS,
+                    self.topic_id))
+            suggestion_thread_dicts = response_dict[
+                'suggestion_thread_dicts']
+
+            self.assertEqual(suggestion_thread_dicts, [])
+
+            feedback_services.create_thread(
+                feconf.ENTITY_TYPE_TOPIC, self.topic_id, self.owner_id,
+                'a subject', 'some text', has_suggestion=True)
+
+            response_dict = self.get_json(
+                '%s/%s' % (
+                    feconf.FEEDBACK_THREADLIST_URL_PREFIX_FOR_TOPICS,
+                    self.topic_id))
+            suggestion_thread_dicts = response_dict[
+                'suggestion_thread_dicts'][0]
+            topic_thread = feedback_services.get_all_threads(
+                feconf.ENTITY_TYPE_TOPIC, self.topic_id, True)[0]
+
+            self.assertEqual(suggestion_thread_dicts['subject'], 'a subject')
+            self.assertEqual(
+                suggestion_thread_dicts['thread_id'], topic_thread.id)
+        self.logout()
+
+
+class RecentFeedbackMessagesHandlerTests(test_utils.GenericTestBase):
+
+    def setUp(self):
+        super(RecentFeedbackMessagesHandlerTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.set_admins([self.OWNER_USERNAME])
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.exp_id = 'exp_id'
+
+    def test_get_recently_posted_feedback_messages(self):
+        self.login(self.OWNER_EMAIL)
+
+        response = self.get_json(
+            feconf.RECENT_FEEDBACK_MESSAGES_DATA_URL)
+
+        self.assertEqual(response['results'], [])
+        self.assertFalse(response['more'])
+
+        self.save_new_valid_exploration(
+            self.exp_id, self.owner_id, title='Exploration title',
+            category='Architecture', language_code='en')
+        feedback_services.create_thread(
```

Comment:
Ummm.... If you are testing the most recent, try and add more than one thread. You will not be able to make out if this is the most recent or not if you just have one thread. This might help increase coverage, but I feel this test is not testing anything useful...

Reply:
Done